### PR TITLE
0.8 port

### PR DESCRIPTION
**Pull request by atotic**

### CODE REVIEW

**Comment by ebidel** [comment]
@jeffposnick can you take a look at this. I took a quick pass and it looks good, but you know the code better.

**Comment by jeffposnick** [comment]
Will do.

**Comment by jeffposnick** [comment]
:+1: This looks great to me—nothing stood out as being a red flag, and the demos look functional. Thanks for working on it, @atotic!

**Comment by atotic** [comment]
Thanks.

The only part where I was not sure what the right thing to do was inside
determinePlaySupported(). The code adds a <video> element for feature
detection, and deletes it when it is done.

Original code added <video> to <google-youtube> element. This does not work
in 0.8 because determinePlaySupported() gets called before element is
ready, and this.root is not inside the body.document yet.

My fix was to add <video> to document.body instead. This breaks
encapsulation a bit, but keeps code simple. Keeping <video> inside
<google-youtube> would involve waiting for both localstorage-loaded and
ready, which means extra logic, etc.

I'll remove my FIXMEs now.

**Comment by jeffposnick** [comment]
That doesn't sound like a huge issue to me, since it's only performed once per browser and the `<video>` element gets cleaned up.

File: bower.json [hunk 1]
```diff
@@ -24,8 +24,11 @@
   "license": "Apache-2.0",
   "homepage": "https://googlewebcomponents.github.io/google-youtube",
   "dependencies": {
-    "core-localstorage": "Polymer/core-localstorage#^0.5.4",
-    "google-apis": "GoogleWebComponents/google-apis#~0.4.3",
-    "polymer": "Polymer/polymer#^0.5.4"
+    "iron-localstorage": "PolymerElements/iron-localstorage#0.8-preview",
+    "google-apis": "GoogleWebComponents/google-apis#0.8-preview",
+    "polymer": "Polymer/polymer#0-8.preview"
```
Comment (christurvey) [inline bower.json]: The import will be bad. The bower path for polymer would be listed as:
"polymer": "Polymer/polymer#0.8-preview"

Comment (atotic) [inline bower.json]: Fixed.

**Comment by ebidel** [comment]
Is this ready? It needs a rebase. I can do that if you want.

**Comment by atotic** [comment]
Ready here.

**Pull request merged**